docker/Dockerfile.ca-styleguide

```diff
@@ -1,23 +1,37 @@
-FROM node:14-alpine
+FROM ruby:2.7-alpine
 
-ENV BUNDLER_VERSION 2.1.4
-# Alpine doesn't keep older patch versions, so we can only specify the minor version here
-ENV RUBY_VERSION=2.6
-ENV LANG=C.UTF-8 \
+ENV NODE_MAJOR_VERSION=14 \
+    BUNDLER_VERSION=2.1.4 \
+    LANG=C.UTF-8 \
     BUNDLE_JOBS=6 \
     BUNDLE_RETRY=3
 
-RUN apk --no-cache --update add ruby~$RUBY_VERSION ruby-dev~$RUBY_VERSION ruby-json ruby-etc gcc libc-dev libxslt-dev libxml2-dev zlib-dev build-base ruby-bigdecimal git && gem install bundler -v $BUNDLER_VERSION
+RUN apk add --no-cache build-base inotify-tools && \
+    gem install bundler -v $BUNDLER_VERSION && \
+    # These next 2 lines essentially do a fancy form of apt-get clean (In alpine)
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
+    truncate -s 0 /var/log/*log && \
+    # This allows us to pull from a different repo than default
+    # Default repo is (3.12), which whilst latest only contains Node 12
+    # This new repo allows us to install Node 14 LTS
+    apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/main/ nodejs~$NODE_MAJOR_VERSION npm~$NODE_MAJOR_VERSION
+
+RUN addgroup --gid 1000 ruby && \
+    adduser --home /home/ruby --uid 1000 --ingroup ruby --disabled-password ruby
 
 RUN mkdir -p /app && chown -R 1000:1000 /app
-COPY --chown=node:node package*.json Gemfile* /app/
+COPY --chown=1000:1000 package*.json Gemfile* /app/
```

Comment:
Out of interest why "1000" for the user id / group?

Reply:
Not a clue, it works 🤷‍♂️

Reply:
Technically it could be ruby:ruby, since the user/group is getting created with those ids, but kubernetes gets very bitchy about using user names in Dockerfiles instead of UIDs. So, to be consistent, it's just easier to use numbers throughout and 1000 seems to be the standard non-root user.

Also, using a non-root user for processes is a really good idea to prevent potential container escape issues.